src/rviz/validate_quaternions.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2017, Stefan Fabian
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Willow Garage, Inc. nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef RVIZ_VALIDATE_QUATERNIONS_H
+#define RVIZ_VALIDATE_QUATERNIONS_H
+
+#include <geometry_msgs/PoseStamped.h>
+#include <OgreQuaternion.h>
+#include <tf/LinearMath/Quaternion.h>
+
+#include <boost/array.hpp>
+
+namespace rviz
+{
+
+inline bool validateQuaternions( float w, float x, float y, float z )
+{
+  return std::abs( w * w + x * x + y * y + z * z - 1.0f ) < 10e-5f;
```

Comment:
Was there a particular reason you switched to using `10e-5`? If not, in tf2 they found the tolerance had to be relaxed it to `10e-3` (https://github.com/ros/geometry2/commit/634fd0cca9d8eb54a2040f5f7f837c9f3156504a), so please stick with `10e-3`

Reply:
Not really, I just felt like 10e-3 was a bit too relaxed for my preferences but if empirical evidence proves me wrong, I'll change it back to 10e-3.